### PR TITLE
Ability to send vpc logs to an external bucket

### PR DESCRIPTION
**Pull request by nikskiz**
# Summary

Ability to send VPC flow logs to an external create bucket. I also added `make readme`

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -262,6 +262,18 @@ variable "flow_log_bucket_name" {
...
 
+variable "send_flow_log_to_external_bucket" {
+  type        = bool
+  description = "Whether to send the flow log to an external created bucket"
+  default     = false
+}
+
+variable "flow_log_external_bucket" {
+  type        = string
+  description = "Name of the external bucket. Note, the bucket will need to allow the account where the VPC resides in as a trustred principa"
```
Comment (aquam8) [inline variables.tf:273]:
```suggestion
  description = "Name of the external bucket. Note, the bucket will need to allow the account where the VPC resides in as a trusted principal"
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Generate markdown documentation for a terraform module (expect SUBFOLDER)
+readme:
+	docker-compose run --rm terra bash -c "sed -i '1,/<!-- TERRAFORM-DOCS-GENERATION -->/!d' README.md; terraform-docs markdown . >> README.md"
+.PHONY: readme
+
+## Generate markdown documentation for all modules
+all-readme:
+	cat .generate-all-readme.sh | docker-compose run --rm terra bash
+.PHONY: all-readme
```
Comment (aquam8) [inline Makefile:37]: This target requires '.generate-all-readme.sh' file to be present Nik

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+.PHONY: fmt
+
+## Validate with tflint
+lint:
+	docker-compose run --rm terra tflint --module
+.PHONY: lint
+
+## Scans all modules for potential security issues
+tfsec:
+	cat .run-tfsec.sh | docker-compose run --rm terra bash
```
Comment (aquam8) [inline Makefile:26]: Same here, `.run-tfsec.sh` is needed, and you may need to adjust path/regex in that file

File: Makefile (hunk 2 above)
Comment (nikskiz) [reply Makefile:37]: Thanks. Since this repo only contains one readme Ill just remove the make target for `all-readme`

File: Makefile (hunk 3 above)
Comment (nikskiz) [reply Makefile:26]: Nice pickup. I have added the bash file and edited the regex

**Pull request merged**